sites/docs/docs/plugins/creating-a-plugin/datasources.md

```diff
@@ -0,0 +1,75 @@
+| `fileFormat`     | 'json' \| 'yaml'                                        | ❌        | If `type` is `file`, set how it should be parsed. It will then be available to `references`                                                                                                         |
+| `description`    | string                                                  | ❌        | Description of the option, shown as a hint in UI                                                                                                                                                    |
+| `children`       | `Record<string\|number\|boolean, Options>`              | ❌        | See [children](#children)                                                                                                                                                                           |
+| `required`       | boolean                                                 | ❌        | Indicates that the user must provide this option                                                                                                                                                    |
+| `options`        | `Array<{label: string, value:string}>`                  | ❌        | Available options for `select` type                                                                                                                                                                 |
+| `nest`           | boolean                                                 | ❌        | Determines behavior of `children`                                                                                                                                                                   |
+| `default`        | 'string' \| 'number' \| 'boolean'                       | ❌        | Default Value                                                                                                                                                                                       |
+
+### Children
+
+Many datasources have variable configuration (e.g. if ssl is enabled for postgres, then an ssl mode can be selected), and Evidence
+options support this workflow.
+
+Consider this partial postgres ssl option:
+
+```javascript
+ssl: {
+    type: 'boolean',
+    // ...
+    nest: true,
+    children: {
+        [true]: {
+            sslmode: {
+                // ...
+            }
+        }
+    }
+},
+```
+
+`ssl.children` is a record of possible values to an additional set of options that are exposed then the values match.
+In this example, the `sslmode` option is only displayed when `ssl` is true.
+
+The resulting type of this option is:
+```typescript
+{ ssl: false } | { ssl: { sslmode: string } }
+```
+
+In cases where you want a flat object, rather than a nested object; set `nest` to false.
+
+This would produce
+
+```typescript
+{ ssl: false } | { ssl: true, sslmode: string }
+```
+
+
+
+
+
+## Simple Interface (Arrays)
```

Comment:
delete?

Reply:
These are currently documented in the template; should we try to document here as well, or just point people to the template?